### PR TITLE
Added Cypress test for validating MM-18041 Guest User Invitation Flow - Verify UI Elements

### PR DESCRIPTION
**Pull request by srkgupta**
#### Summary
Validating the UI Elements of the Guest User Invitation Flow when Guest Account feature is enabled. Please check the ticket for more details.

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-18041

### CODE REVIEW

File: e2e/cypress/integration/enterprise/guest_accounts/guest_invitation_ui_spec.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        });
+
+        // *Verify Set Custom Message before clicking on the link
+        cy.getByTestId('customMessage').should('be.visible').within(($el) => {
+            cy.wrap($el).get('#customMessageHeader').should('not.exist');
+            cy.wrap($el).get('textarea').should('not.exist');
+            cy.wrap($el).get('.help-text').should('have.text', 'Create a custom message to make your invite more personal.');
+
+            // #Verify link text and click on it
+            cy.wrap($el).get('a').should('have.text', 'Set a custom message').click();
```
Comment (saturninoabril) [inline e2e/cypress/integration/enterprise/guest_accounts/guest_invitation_ui_spec.js]: nit: Can we have href check as well?

Comment (srkgupta) [reply e2e/cypress/integration/enterprise/guest_accounts/guest_invitation_ui_spec.js]: Any reason we need to have this? We are anyways trying to click on it and verify the behavior.

Comment (saturninoabril) [reply e2e/cypress/integration/enterprise/guest_accounts/guest_invitation_ui_spec.js]: 0/5, no particular reason.  We're using anchor element in many ways: opening a link, direct to internal target, etc. and for some of those, we're checking href.

Comment (srkgupta) [reply e2e/cypress/integration/enterprise/guest_accounts/guest_invitation_ui_spec.js]: Thanks. It makes sense to validate href for external links. But for onclick actions, it's better to validate the behavior when a link is clicked.

**Pull request merged**